src/main/java/dev/Roach/StoresFetcher.java

```diff
@@ -0,0 +1,40 @@
+package dev.Roach;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+public class StoresFetcher {
+    private final HttpClient client;
+
+    public StoresFetcher(HttpClient client) {
+        this.client = client;
+    }
+
+    public String getAllShops() {
+        try {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create("https://www.cheapshark.com/api/1.0/stores"))
+                    .GET()
+                    .build();
+
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+
+            int statusCode = response.statusCode();
+
+            if(statusCode == 404) {
+                return "API endpoint or resource you are trying to access does not exist";
```

Comment:
Can this happen actually? This method does not allow change in the URL that is being called, so the only instance where it can happen is if you make a mistake, not the user.

Reply:
I thought about situation when API is down or link was changed but I don't know if this is proper way of handling it. Should I delete this ?

Reply:
As soon as you introduce datamodel here, it won't really make sense. I would say it's safe to delete this.